### PR TITLE
Added Eng blog to habitat.sh site

### PR DESCRIPTION
**Pull request by eeyun**
Ok,

It's not a totally massive addition, but I'm totally not a web developer. If anyone can pull this down and build it locally and check it out that would be good. I've verified its a reasonable experience. 

As there are no blog articles in this PR in order to verify how all of the things look with an actual blog post you'll need to add some markdown files to the www/source/blog directory.

If there are any glaringly obscene things in my css or slim templates please let me know.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by eeyun** [comment]
@thesentinels deploy_acceptance

**Comment by eeyun** [comment]
@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/222094341).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/222094341).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

File: www/source/blog/tag.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+pageable: true
+per_page: 12
+---
+<h1>Articles tagged '<%= tagname %>'</h1>
```
Comment (cnunciato) [inline www/source/blog/tag.html.erb]: Layout looks a little off, here -- I think we want this inside a `container blog` as well, like the index page, or to give it a similar container specific to tags.

Also, should we make these new views `.slim`s also? Seems like we should go with one or the other. (I don't have a preference on this, other than consistency.)

File: www/source/blog/index.html.slim [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+pageable: true
+per_page: 10
+---
+.container.blog
+  p.pagination-link.small-2.columns
+    = link_to 'RSS', "blog/feed.xml"
+    br
+    = link_to 'Twitter', "https://twitter.com/habitatsh"
+    br
+    = link_to 'Youtube', "https://www.youtube.com/channel/UC0wJZeP2dfPZaDUPgvpVpSg"
```
Comment (cnunciato) [inline www/source/blog/index.html.slim]: Minor: `YouTube`

**Comment by robbkidd** [comment]
A couple of the vim temp files got sucked up into the commits.

```
www/source/blog/partials/._author_bio.html.slim.swp
www/source/blog/partials/.author_bio.html.slim.swp
```

**Comment by raskchanky** [comment]
@robbkidd @eeyun It's probably worth adding `*.swp` to the `.gitignore`.

**Comment by robbkidd** [comment]
@ryankeairns I jammed a list of tags with article counts into the blog index's sidebar. What's still left, I think, is making the page generated by `tag.html.slim` have a sidebar, too. It doesn't use the blog layout at the moment.

**Comment by robbkidd** [comment]
Also, somehow removing the `.swp` files left for author_bio (as in the diff below) means I can't build the site. 

```
/Users/rkidd/work/habitat/www/.direnv/ruby/gems/middleman-core-4.1.2/lib/middleman-core/sitemap/extensions/proxies.rb:89:in `target_resource': 
Path blog/2017/04/11.html proxies to unknown file blog/calendar.html:[long, list, of, pages]
{stacktrace}
```

I am super confused by that.

```diff
diff --git a/.gitignore b/.gitignore
index 52610778..b583e5b2 100644
--- a/.gitignore
+++ b/.gitignore
@@ -8,6 +8,7 @@ log/
 results/
 tmp/
 .vagrant/
+*.swp
 *.sig.key
 *.box.key
 *.sym.key
diff --git a/www/source/blog/partials/._author_bio.html.slim.swp b/www/source/blog/partials/._author_bio.html.slim.swp
deleted file mode 100644
index 3129f1c6..00000000
Binary files a/www/source/blog/partials/._author_bio.html.slim.swp and /dev/null differ
diff --git a/www/source/blog/partials/.author_bio.html.slim.swp b/www/source/blog/partials/.author_bio.html.slim.swp
deleted file mode 100644
index 7702b381..00000000
Binary files a/www/source/blog/partials/.author_bio.html.slim.swp and /dev/null differ
```

**Comment by ryankeairns** [comment]
@robbkidd thanks! I'm hoping to style things up starting Monday (possibly tomorrow afternoon). Unfortunately, I'm of no help for the .swp error :/

**Comment by ryankeairns** [comment]
🚧  After playing with a variety of middlemanapp blog setting in the config.rb, ou can build from this branch again... but it's still not quite 'right'. The most glaring issue is that the /blog/index.html is not spitting out the article summaries.

**Comment by ryankeairns** [comment]
😎  The blog config settings are sorted now, moving on to styles.

**Comment by ryankeairns** [comment]
This is super close to done... the middleman blog settings in config.rb are maddening. It works fine for several restarts then randomly fails to build. It's the last bit to sort out...

**Comment by ryankeairns** [comment]
@eeyun Here's what I've confirmed... the site builds successfully if you are on a ruby version of 2.2.x (specifically I've used 2.2.1 and 2.2.4). The build fails with the following error if you're on 2.3.1 or 2.4.0:

`/Users/ryankeairns/.rvm/gems/ruby-2.3.1/gems/middleman-core-4.1.2/lib/middleman-core/sitemap/extensions/proxies.rb:89:in `target_resource': Path blog/2017/04/11.html proxies to unknown file blog/calendar.html`

If you mess with the blog settings in config.rb it will run occasionally on those newer versions of ruby but it's very unreliable and unpredictable.

The README for middleman-blog states that it supports 2.2.4, 2.3.1, and 2.4.0, but I also see a commit in March of 2017 that updates the travis.yml to include the 2.3 and 2.4 versions... which makes me question how much they've been tested/used to this point: https://github.com/middleman/middleman-blog

My ruby/rails-fu is at best novice, but do you think we should just pin this ruby version (2.2.4 or ~>2.2.0 or something like that) to the Gemfile?

I'll also open an issue on the middleman forum for this :)

cc:/ @chrisnunciato

**Comment by ryankeairns** [comment]
I've gone ahead added `ruby "2.2.4"` to the Gemfile... I presume that will force the build process to use this version of ruby.

**Comment by cnunciato** [comment]
👍 on using 2.2.4. We should also update `www/.ruby-version`, the www section in `.travis.yml`, and the `www/README.md`. That should do it!

**Comment by eeyun** [comment]
Do we for sure want to pin it? Or do we want to try and run down the source of the issue on more recent versions of ruby? I'm asking because I legitimately don't know but if its something we can sort out and be able to float the version I feel like that might be preferable?

**Comment by robbkidd** [comment]
Pins hurt later.

**Comment by cnunciato** [comment]
Definitely preferable yeah, but it's also fine (IMO) to go with what works now and investigate upgrading as a separate task.

**Comment by robbkidd** [comment]
I'll spend a wee while fiddling with upgrades to the middleman component gems.

**Comment by robbkidd** [comment]
OK. Got build and server running with some updated gems. (Blog looks super purdy!)

**Comment by cnunciato** [comment]
Yay, thanks @robbkidd!

**Comment by robbkidd** [comment]
Pushed with the gem version updates.

The only blog page(s) I see that are a bit wacky are the ones generated through the calendar template, e.g. http://localhost:4567/blog/2017/04/ when running the site locally.

**Comment by ryankeairns** [comment]
Woo! Thanks everybody!!

We're not currently linking to the calendar views anywhere (that I'm aware of), so I think we're ok to leave those as-is for the time being.

**Comment by eeyun** [comment]
If your change @robbkidd puts us in a stable place with an upgrade path forward and the rest of the things are done I say we remove the do not merge tag.

**Comment by robbkidd** [comment]
Are these the correct articles and authors for them that we go live with for the new blog?

**Comment by eeyun** [comment]
No apparently not. Something must have gotten muddled. The both of the blog posts were authored by me. Once we launch with the functionality we can add some more content pretty quickly, have some things sitting off to the side.

**Comment by ryankeairns** [comment]
I changed the author on one just to test that it worked, forgot to switch it back.

**Comment by eeyun** [comment]
@ryankeairns I can switch this back in a few minutes unless you get to it before me.

**Comment by ryankeairns** [comment]
@eeyun author switched back to you! Also, I had to add `classes:` in the frontmatter to get the styling of things sorted out during the course of updating things, so I've added it to your Hello World post.

**Comment by tashimi** [comment]
blog

**Comment by eeyun** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/231911826).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/231911826).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**